File: py/src/imbrace/environments.py
from __future__ import annotations
from typing import Literal, Optional
from dataclasses import dataclass, field

Environment = Literal["develop", "sandbox", "stable"]


@dataclass
class ServiceHosts:
    ips: Optional[str] = None
    data_board: Optional[str] = None


@dataclass
class EnvironmentPreset:
    gateway: str
    service_hosts: ServiceHosts = field(default_factory=ServiceHosts)


ENVIRONMENTS: dict[str, EnvironmentPreset] = {
    "develop": EnvironmentPreset(
        gateway="https://app-gateway.dev.imbrace.co",
        service_hosts=ServiceHosts(
            ips="http://ips.dev.imbrace.lan",
            data_board="http://data-board.dev.imbrace.lan",
        ),
    ),
    "sandbox": EnvironmentPreset(
        gateway="https://app-gateway.sandbox.imbrace.co",
    ),
    "stable": EnvironmentPreset(
        gateway="https://app-gateway.imbrace.co",
    ),
}

File: ts/src/environments.ts
export type Environment = 'develop' | 'sandbox' | 'stable'

export interface EnvironmentPreset {
  gateway: string
  /** Per-service host overrides (khi service có dedicated host riêng, không qua gateway) */
  serviceHosts?: {
    /** IPS service — dev dùng ips.dev.imbrace.lan trực tiếp */
    ips?: string
    /** Data Board service — dev dùng data-board.dev.imbrace.lan trực tiếp */
    dataBoard?: string
  }
}

export const ENVIRONMENTS: Record<Environment, EnvironmentPreset> = {
  develop: {
    gateway: 'https://app-gateway.dev.imbrace.co',
    serviceHosts: {
      ips:       'http://ips.dev.imbrace.lan',
      dataBoard: 'http://data-board.dev.imbrace.lan',
    },
  },
  sandbox: {
    gateway: 'https://app-gateway.sandbox.imbrace.co',
  },
  stable: {
    gateway: 'https://app-gateway.imbrace.co',
  },
}
